src/sentry/auth/helper.py

```diff
@@ -379,6 +382,19 @@ def _post_login_redirect(self) -> HttpResponseRedirect:
 
         return response
 
+    def _check_verification_key(self, identity):
+        if self.request.session.get("confirm_account_verification_key"):
+            verification_value = get_verification_value_from_key(
+                self.request.session["confirm_account_verification_key"]
+            )
+            acting_user = self._get_user(identity)
+            if (
+                verification_value["email"] == identity["email"]
+                and verification_value["user_id"] == acting_user.id
```

Comment:
did we want to check the member id here too?

Reply:
we could, we'd have to query OrganizationMember since i don't think we have the member id available anywhere since ` has_membership = OrganizationMember.objects.filter(
                user=acting_user, organization=self.organization
            ).exists()`
is within the if statement after this check. I'll add the check if we think its needed

Reply:
hmm thinking through edge cases, I think we're okay not to have a check here. will keep that going in the background but fine as is for now.